nixos/modules/programs/mininet.nix

```diff
@@ -0,0 +1,33 @@
+# Global configuration for mininet
+# kernel must have NETNS/VETH/SCHED
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg  = config.programs.mininet;
+
+  pyEnv = pkgs.python.withPackages(ps: [ ps.mininet-python ]);
+in
+{
+  options.programs.mininet.enable = mkEnableOption ''
+      Whether to enable mininet.
+    '';
+
+  config = mkIf cfg.enable {
+
+    virtualisation.vswitch.enable = true;
+
+    environment.systemPackages = with pkgs; [
```

Comment:
I wonder. I could write a wrapper for the official "mn" script but what about user scripts ? users would need to add these tools (iperf & co) themselves to PATH. People using mininet are likely to either use it in a VM or playing with iperf either way.

Reply:
Didn't they first attach to network namespaces with a certain command, then the PATH would be inherited in this environment, no? I has been a while since I used it.

Reply:
Here is where they mount the network namespace https://github.com/teto/mininet/blob/9a069563e1f4fb5b544249a05fe5557416183827/mnexec.c#L129, I am not sure how PATH behave wit namespaces. Commands are run by this same mnexec binary so I believe wrapping it should work. Will try this evening.